core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -820,10 +800,38 @@ private void failTask(Throwable cause)
         taskStatusFetcher.updateTaskStatus(status);
 
         try (SetThreadName ignored = new SetThreadName("HttpRemoteTask-%s", taskId)) {
-            scheduleAsyncCleanupRequest(new Backoff(maxErrorDuration), "abort", true);
+            if (cause instanceof TrinoTransportException) {
+                // task is unreachable
+                cleanUpLocally();
```

Comment:
can `TrinoTransportException` be transient?
would be reasonable to still send abort to the task (eg async-ly, or with a `try`)?

Reply:
All communication requests are retried multiple times. By default requests are retied for 5 minutes: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java#L58. When a task fails to respond for that long it feels like it is rather unlikely for it to respond in the following 5 minutes.

But I aggre, there's always a chance it might. However, currently tasks are configured to expire in 2 minutes  (https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/TaskManagerConfig.java#L62) and the implications of not sending an abort request are rather low (the task is going to be expired anyway and garbage collected anyway).